File: 1029.Two City Scheduling.py
class Solution(object):
    def twoCitySchedCost(self, costs):
        """
        :type costs: List[List[int]]
        :rtype: int
        """
        diff = [(cost[0] - cost[1], cost) for cost in costs]
        diff.sort()
        N = len(costs) / 2
        return sum(pair[1][0] for pair in diff[:N]) + sum(pair[1][1] for pair in diff[N:])

File: 1029.Two City Scheduling.rb
# @param {Integer[][]} costs
# @return {Integer}
def two_city_sched_cost(costs)
    costs.sort_by { |cost| cost[0] - cost[1] }
        .each_with_index
        .map { |cost, index| index < costs.size / 2 ? cost[0] : cost[1] }
        .sum
end
